### ISSUE TITLE
VadYaraScan URL error

### ISSUE THREAD

**Issue by AsafEitani** [volatilityfoundation/volatility3#408]
**Describe the bug**
Getting "unknown url type: c" when providing the full path to yara file:

```
C:\ProgramData\Anaconda3\python.exe C:\Users\user\volatility3\vol.py -vvv -r json -f C:/Users/user/Desktop/research/Memory/dump.vmem windows.vadyarascan.VadYaraScan --yara-file "C:\Users\user\Documents\yara_signatures.yar"


Volatility 3 Framework 2.0.0-beta.1
INFO     root        : Volatility plugins path: ['C:\\Users\\user\\volatility3\\volatility\\plugins', 'C:\\Users\\user\\volatility3\\volatility\\framework\\plugins']
INFO     root        : Volatility symbols path: ['C:\\Users\\user\\volatility3\\volatility\\symbols', 'C:\\Users\\user\\volatility3\\volatility\\framework\\symbols']
INFO     volatility.framework.automagic: Detected a windows category plugin
INFO     volatility.framework.automagic: Running automagic: ConstructionMagic
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary
Level 9  volatility.framework.automagic.construct_layers: Failed on requirement: plugins.VadYaraScan.primary
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary
Level 9  volatility.framework.automagic.construct_layers: Failed on requirement: plugins.VadYaraScan
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.automagic.construct_layers: Failed on requirement: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.automagic.construct_layers: Failed on requirement: plugins.VadYaraScan
Level 9  volatility.framework.interfaces.configuration: TypeError - yara_rules requirements only accept str type: None
Level 9  volatility.framework.interfaces.configuration: TypeError - yara_rules requirements only accept str type: None
INFO     volatility.framework.automagic: Running automagic: LayerStacker
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
DEBUG    volatility.framework.automagic.windows: DTB was found at: 0x1a8000
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary
Level 9  volatility.framework.configuration.requirements: IndexError - No configuration provided: plugins.VadYaraScan.primary.memory_layer
Level 9  volatility.framework.interfaces.configuration: TypeError - kernel_virtual_offset requirements only accept int type: None
Level 9  volatility.framework.interfaces.configuration: TypeError - kernel_virtual_offset requirements only accept int type: None
Level 9  volatility.framework.interfaces.configuration: TypeError - kernel_banner requirements only accept str type: None
Level 9  volatility.framework.interfaces.configuration: TypeError - kernel_banner requirements only accept str type: None
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.automagic.construct_layers: Failed on requirement: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.automagic.construct_layers: Failed on requirement: plugins.VadYaraScan
Level 9  volatility.framework.interfaces.configuration: TypeError - yara_rules requirements only accept str type: None
Level 9  volatility.framework.interfaces.configuration: TypeError - yara_rules requirements only accept str type: None
DEBUG    volatility.framework.automagic.stacker: Stacked layers: ['IntelLayer', 'FileLayer']
INFO     volatility.framework.automagic: Running automagic: WinSwapLayers
INFO     volatility.framework.automagic: Running automagic: WintelHelper
INFO     volatility.framework.automagic: Running automagic: KernelPDBScanner
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
Level 9  volatility.framework.configuration.requirements: Symbol table requirement not yet fulfilled: plugins.VadYaraScan.nt_symbols
DEBUG    volatility.framework.automagic.pdbscan: Kernel base determination - searching layer module list structure
DEBUG    volatility.framework.symbols.windows.pdbutil: Using symbol library: ntkrpamp.pdb\C6401D4EE1EC427283A8FA17771EFB53-1
DEBUG    volatility.framework.automagic.pdbscan: Setting kernel_virtual_offset to 0x81c0b000
Progress:  100.00               PDB scanning finished
Traceback (most recent call last):
  File "C:\Users\user\volatility3\vol.py", line 10, in <module>
    volatility.cli.main()
  File "C:\Users\user\volatility3\volatility\cli\__init__.py", line 582, in main
    CommandLine().run()
  File "C:\Users\user\volatility3\volatility\cli\__init__.py", line 315, in run
    renderers[args.renderer]().render(constructed.run())
  File "C:\Users\user\volatility3\volatility\cli\text_renderer.py", line 351, in render
    grid.populate(visitor, final_output)
  File "C:\Users\user\volatility3\volatility\framework\renderers\__init__.py", line 211, in populate
    for (level, item) in self._generator:
  File "C:\Users\user\volatility3\volatility\framework\plugins\windows\vadyarascan.py", line 53, in _generator
    rules = yarascan.YaraScan.process_yara_options(dict(self.config))
  File "C:\Users\user\volatility3\volatility\framework\plugins\yarascan.py", line 80, in process_yara_options
    rules = yara.compile(file = resources.ResourceAccessor().open(config['yara_file'], "rb"))
  File "C:\Users\user\volatility3\volatility\framework\layers\resources.py", line 108, in open
    raise excp
  File "C:\Users\user\volatility3\volatility\framework\layers\resources.py", line 94, in open
    fp = urllib.request.urlopen(url, context = self._context)
  File "C:\ProgramData\Anaconda3\lib\urllib\request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "C:\ProgramData\Anaconda3\lib\urllib\request.py", line 525, in open
    response = self._open(req, data)
  File "C:\ProgramData\Anaconda3\lib\urllib\request.py", line 548, in _open
    'unknown_open', req)
  File "C:\ProgramData\Anaconda3\lib\urllib\request.py", line 503, in _call_chain
    result = func(*args)
  File "C:\ProgramData\Anaconda3\lib\urllib\request.py", line 1387, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: c>
```

**Context**
Volatility Version: 2.0.0-beta.1
Operating System:  Windows 10
Python Version: 3.6.9
Suspected Operating System: Windows 8.1
Command:  `C:\ProgramData\Anaconda3\python.exe C:\Users\user\volatility3\vol.py -vvv -r json -f C:/Users/user/Desktop/research/Memory/dump.vmem windows.vadyarascan.VadYaraScan --yara-file "C:\Users\user\Documents\yara_signatures.yar"`

**To Reproduce**

1. use vadyarascan with a full path to file
2. See error

**Comment by ikelos** [comment]
Yep, this is because volatility tries to interpret it as a URL.  On windows you need to use `\\` rather than just `\` on the command line.  I'll try to improve the error messaging though...

**Comment by AsafEitani** [comment]
But python should receive the string escaped... like when you pass `-f` the path it will escape it automatically....

**Comment by ikelos** [comment]
You're right, sorry, escaping the slashes isn't the problem.  The parameter takes a URL and apparently it's not figuring out that it's a local file (and prepending file:// automatically).  You can work around the issue by prepending `file://` and I'll work on the implementation and see if I can find a solution...

**Comment by ikelos** [comment]
This issue should have been resolved by #411 which has been merged.  Please feel free to reopen it if you're still encountering difficulties...
